x-pack/plugins/security_solution/server/lib/detection_engine/rule_actions/legacy_types.ts

```diff
@@ -26,8 +26,18 @@ export interface IRuleActionsAttributes extends Record<string, any> {
  * We keep this around to migrate and update data for the old deprecated rule actions saved object mapping but we
  * do not use it anymore within the code base. Once we feel comfortable that users are upgrade far enough and this is no longer
  * needed then it will be safe to remove this saved object and all its migrations.
- * @deprecated
+ * @deprecated Remove this once the legacy notification/side car is gone
  */
-export interface IRuleActionsAttributesSavedObjectAttributes
-  extends IRuleActionsAttributes,
+export interface LegacyIRuleActionsAttributesSavedObjectAttributes
+  extends LegacyIRuleActionsAttributes,
     SavedObjectAttributes {}
+
+/**
+ * @deprecated Remove this once the legacy notification/side car is gone
+ */
+export interface LegacyRuleActions {
+  id: string;
+  actions: RuleAlertAction[];
+  ruleThrottle: string;
```

Comment:
This predates this PR but I'm still a bi confused with `ruleThrottle` and `alertThrottle`. Shouldn't these match? Did we figure out why we have both?

Reply:
Yeah, they're weird. Only `ruleThrottle` exists when you set the rule to execute immediately. We saw that on your computer from Friday when testing which was our surprise. There is still a side car saved object but it only has `ruleThrottle` set to `no_action` in that instance. Otherwise it is set to `1h`, `1d`, etc... The one below it can be set or not set.

I know that's bizarre but that's the legacy behavior so this type is correct. You can verify this on your 7.14 system if you want again but that is what we saw.